### PR TITLE
feat: Update the `RouteConfiguration` class to include the `parameters` field

### LINKED ISSUE #1450: feat: Update the `RouteConfiguration` class to include the `parameters` field
**Description**
Update the `RouteConfiguration` class to include the `parameters` field. Implement

**Acceptance Criteria**
- [ ] Add new field to the `RouteConfiguration` class
- [ ] Implement the `RouteConfigurationLocationConverter`
- [ ] Update the `MetricsRouteInformationParser` to use the `RouteConfigurationLocationConverter`
- [ ] Test new functionality

**Additional context**
[Deep Links document](https://github.com/Flank/flank-dashboard/blob/master/metrics/web/docs/features/deep_links/01_deep_linking.md)

### PR DESCRIPTION
**Pull request by solid-daniyilyevtyushkin**
Fixes #1450

## Checklist

- [x] Documented
- [x] Unit tested

### CODE REVIEW

File: metrics/web/lib/common/presentation/navigation/route_configuration/route_configuration_location_converter.dart [hunk 1]
```diff
@@ -0,0 +1,28 @@
+// Use of this source code is governed by the Apache License, Version 2.0
+// that can be found in the LICENSE file.
+
+import 'package:metrics/common/presentation/navigation/route_configuration/route_configuration.dart';
+
+/// A class that provides a method for converting the given [RouteConfiguration]
+/// to a location [String].
```
Comment (solid-vovabeloded) [inline metrics/web/lib/common/presentation/navigation/route_configuration/route_configuration_location_converter.dart:7]: 🏷️
```suggestion
/// A class that provides a method for converting the [RouteConfiguration]
/// to a location [String].
```
What do you think?

File: metrics/web/test/common/presentation/navigation/metrics_route_information_parser_test.dart [hunk 2]
```diff
@@ -87,14 +97,26 @@ void main() {
     );
 
     test(
-      ".restoreRouteInformation() returns the route information with the location equals to the given route configuration path",
+      ".restoreRouteInformation() delegates converting the route configuration to a location to the route configuration location converter if it is not null",
```
Comment (solid-vovabeloded) [inline metrics/web/test/common/presentation/navigation/metrics_route_information_parser_test.dart:100]: 🏷️
```suggestion
      ".restoreRouteInformation() delegates converting the route configuration to the route configuration location converter if the configuration is not null",
```
Would it be a bit more clear?

**Review by solid-vovabeloded — CHANGES_REQUESTED** [review]
Great job, @solid-daniyilyevtyushkin & @solid-yevheniiashchypeleva! I've added a couple of suggestions here - please take a look. Overall, the code looks clean and well-structured - thanks!

**Review by solid-vladlaba — CHANGES_REQUESTED** [review]
Well done, @solid-daniyilyevtyushkin @solid-yevheniiashchypeleva! I've added a few suggestions below - please let me know what you think!

**Review by solid-vladlaba — APPROVED** [review]
LGTM, thanks @solid-daniyilyevtyushkin @solid-yevheniiashchypeleva!:rocket:

**Review by solid-vovabeloded — APPROVED** [review]
LGTM, thanks @solid-daniyilyevtyushkin & @solid-yevheniiashchypeleva!

**Pull request merged**